src/libstd/io/stdio.rs

```diff
@@ -872,6 +872,91 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Equivalent to the [`input`] method except that
+/// a prompt is printed before capturing input.
+///
+/// # Note
+///
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+///
+/// # Panics
+///
+/// This function currently panics if it does not receive input,
+/// or when input is empty without newline
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(prompt)]
+/// use std::io::prompt;
+///
+/// fn main() {
+///     let name = prompt("Enter name: ").expect("input failed!");
+///
+///     println!("Your name is {}!", name);
+/// }
+/// ```
+#[unstable(
+    feature = "prompt",
+    reason = "this function may be replaced with a \
+                     more general mechanism",
+    issue = "none"
+)]
+pub fn prompt(prompt: &str) -> Result<String> {
+    stdout().write(prompt.as_bytes()).expect("failed to write to stdout");
+    Ok(input()?)
```

Comment:
I don't think we should panic given that we propagate the error, user can `unwrap` themselves, we don't need to take this burden. Also, the docs should be updated.

```suggestion
    stdout().write(prompt.as_bytes())?;
    input()
```